### PR TITLE
Implemented a way to verify webhook signature

### LINKED ISSUE #321: Unable to verify/handle webhooks with the new API
`Mailgun::verifyWebhookSignature()` is deprecated and will be removed in 3.0. `Mailgun\Api\Webhook` does not offer this functionality yet. It might be worth investigating if the received events should be parsed and wrapped in models too.

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #321 
I just moved the function to the Webhook API

### CODE REVIEW

File: src/Mailgun/Mailgun.php [hunk 1]
```diff
@@ -84,20 +84,20 @@ public function __construct(
...
-        HttpClientConfigurator $httpClientConfigurator,
+        HttpClientConfigurator $configurator,
         Hydrator $hydrator = null,
         RequestBuilder $requestBuilder = null
     ) {
-        $httpClient = $httpClientConfigurator->createConfiguredClient();
+        $httpClient = $configurator->createConfiguredClient();
 
-        return new self(null, $httpClient, 'api.mailgun.net', $hydrator, $requestBuilder);
+        return new self($configurator->getApiKey(), $httpClient, 'api.mailgun.net', $hydrator, $requestBuilder);
```
Comment (DavidGarciaCat) [inline src/Mailgun/Mailgun.php]: Why this change? It was not here until now, and I don't see calls to this static method (or I am too tired).

Comment (Nyholm) [inline src/Mailgun/Mailgun.php]: The webhook API need the API key to be able to verify the signature.
